solr/test-framework/src/java/org/apache/solr/embedded/JettySolrRunner.java

```diff
@@ -556,15 +528,7 @@ public synchronized void start(boolean reusePort) throws Exception {
           server.start();
         }
       }
-      synchronized (JettySolrRunner.this) {
```

Comment:
after server.start returns, dispatchFilter should be running.  It's not started asynchronously; no race.